### PR TITLE
showcase/starter: stop dev logging in 'production' mode

### PR DESCRIPTION
**Pull request by SuperITMan**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[X] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

All the logs displayed in 'dev' mode in the showcase are also displayed in the 'prod' build.

Since we are going to release our final release, we should have a clean console without all those logs.

## What is the new behavior?

Almost no logs in 'production' build of the "showcase" and the "starter".

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

## Other information

### CODE REVIEW

File: showcase/src/app/app.module.ts [hunk 1]
```diff
@@ -120,7 +121,7 @@ export function initRouterLog(router: UIRouter): () => void {
 export function getXsrfWaitBeforePinging(sessionService: StarkSessionService): Observable<any> {
 	let waitFor$: Observable<any> = of("production"); // no need to wait on production
 
-	if (ENV !== "production") {
+	if (ENV === "development") {
```
Comment (nicanac) [inline showcase/src/app/app.module.ts]: But what will happen in UAT, don't wee need the logger from time to time ?

Comment (nicanac) [reply showcase/src/app/app.module.ts]: Maybe they can just setup their applicationConfig (in uat), like the one of dev. 
In this case, my remark is useless :-)

Comment (SuperITMan) [reply showcase/src/app/app.module.ts]: Actually, in the case of the showcase, we don't have any reason to have a "UAT" logic.

If we wanted to manage the different NBB environments, we could add some properties related to the environments we have in "stark-core/src/common/common/environment/environment.intf.ts" or the apps themselves could extend `StarkEnvironment` and add the needed environment, as explained here: [ENVIRONMENTS.md#how-to-add-more-properties-to-the-environment-file](https://github.com/NationalBankBelgium/stark/blob/master/docs/stark-core/ENVIRONMENTS.md#how-to-add-more-properties-to-the-environment-file)

In any case, I think this should be part of another issue. Could you please create an issue "type: discussion / RFC" with ? :blush:

**Comment by SuperITMan** [comment]
@nicanac I created 2 different commits to have a commit message related to each one, like this both will be present in the CHANGELOG. What do you think ? :blush:

File: showcase/src/app/app.module.ts (hunk 1 above)
Comment (nicanac) [reply showcase/src/app/app.module.ts]: Ye sure, maybe it's useless, I don't know. 
But, it could be a small improvement.

Comment (nicanac) [reply showcase/src/app/app.module.ts]: > @nicanac I created 2 different commits to have a commit message related to each one, like this both will be present in the CHANGELOG. What do you think ? 😊

Ok, let's keep it like that. It'll add some substance to the change log :-)

**Pull request merged**